solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -236,16 +242,19 @@ protected ZkCredentialsProvider createZkCredentialsToAddAutomatically() {
     if (!StringUtils.isEmpty(zkCredentialsProviderClassName)) {
       try {
         log.info("Using ZkCredentialsProvider: {}", zkCredentialsProviderClassName);
-        return (ZkCredentialsProvider)
-            Class.forName(zkCredentialsProviderClassName).getConstructor().newInstance();
+        ZkCredentialsProvider zkCredentialsProvider =
+            (ZkCredentialsProvider)
+                Class.forName(zkCredentialsProviderClassName).getConstructor().newInstance();
+        zkCredentialsProvider.setZkCredentialsInjector(zkCredentialsInjector);
+        return zkCredentialsProvider;
       } catch (Throwable t) {
         // just ignore - go default
         log.warn(
             "VM param zkCredentialsProvider does not point to a class implementing ZkCredentialsProvider and with a non-arg constructor",
             t);
       }
     }
-    log.debug("Using default ZkCredentialsProvider");
+    log.warn("Using default ZkCredentialsProvider");
```

Comment:
Why is using the default provider bad? Is that one insecure?

Making this a _warn_ level message should be accompanied by either a link to the docs or suggestions on what to do. Otherwise operators are going to see the message and promptly ignore it.

Reply:
The default one is not secure, it defaults the ACLs to _OPEN_ACL_UNSAFE_
Will add some explanation/suggestions to the message.